p2p/libp2p/netMessenger.go

```diff
@@ -748,31 +739,33 @@ func (netMes *networkMessenger) Broadcast(topic string, buff []byte) {
 }
 
 // RegisterMessageProcessor registers a message process on a topic
-func (netMes *networkMessenger) RegisterMessageProcessor(topic string, handler p2p.MessageProcessor) error {
+func (netMes *networkMessenger) RegisterMessageProcessor(topic string, identifier string, handler p2p.MessageProcessor) error {
 	if check.IfNil(handler) {
 		return p2p.ErrNilValidator
 	}
 
 	netMes.mutTopics.Lock()
 	defer netMes.mutTopics.Unlock()
-	validator := netMes.processors[topic]
-	if !check.IfNil(validator) {
-		return fmt.Errorf("%w, operation RegisterMessageProcessor, topic %s",
-			p2p.ErrTopicValidatorOperationNotSupported,
-			topic,
-		)
+	topicProcs := netMes.processors[topic]
+	if topicProcs == nil {
+		topicProcs = newTopicProcessors()
+		netMes.processors[topic] = topicProcs
+
+		err := netMes.pb.RegisterTopicValidator(topic, netMes.pubsubCallback(topicProcs, topic))
```

Comment:
Just to make sure I understand the changes correctly.
Previously: the topicMessageProcessor acts also as a validator for the messages, so one message will not be propagated to the network if the validator returned an error.

Now: consider the following scenario where we have two processors on the same topic, one of them does in-depth validation and returns an error on one message in case e.g aggregated signature is invalid/cannot be validated, but the other doesn't care about signatures and only does some statistics on the received messages. 

1. Will the node propagate the message to the network or not if there is one processor returning an error and another returning "no error"? 

2. In case both processors return "no error" will the node propagate the message twice or only once?

Can we also add some tests for the above?

Reply:
I saw below that you do not propagate the message if either of the processors return an error, so that answers my first question.

Also for my second question, I see you only return from pubsub callback when both processors have finished, so if both returned no error then message is propagated only once

:+1:

Reply:
Right, all registered processors are called. If at least one of them returns an error, the message won't propagate to the other peers. If all processors return nil errors, the message will propagate only once to the other peers.